common/djangoapps/student/views.py

```diff
@@ -1859,16 +1867,48 @@ def _enroll_user_in_pending_courses(student):
                 )
 
 
-def _record_registration_attribution(request, user):
+def record_affiliate_registration_attribution(request, user):
     """
     Attribute this user's registration to the referring affiliate, if
     applicable.
     """
     affiliate_id = request.COOKIES.get(settings.AFFILIATE_COOKIE_NAME)
-    if user is not None and affiliate_id is not None:
+    if user and affiliate_id:
         UserAttribute.set_user_attribute(user, REGISTRATION_AFFILIATE_ID, affiliate_id)
 
 
+def record_utm_registration_attribution(request, user):
+    """
+    Attribute this user's registration to the latest UTM referrer, if
+    applicable.
+    """
+    utm_cookie_name = RegistrationCookieConfiguration.current().utm_cookie_name
+    utm_cookie = request.COOKIES.get(utm_cookie_name)
+    if user and utm_cookie:
+        utm = json.loads(utm_cookie)
+        for utm_parameter in REGISTRATION_UTM_PARAMETERS:
```

Comment:
You can do `for key, value in REGISTRATION_UTM_PARAMETERS.items()`. Maybe with better names.

I wonder if you can modify the dictionary so it contains callables that accept a dictionary and extract the relevant UTM fields.